crates/bevy_ecs/src/hierarchy.rs

```diff
@@ -0,0 +1,345 @@
+// TODO: REMOVE THIS
+#![allow(missing_docs)]
+
+//! Parent-Child relationships for entities.
+
+use crate as bevy_ecs;
+use crate::bundle::Bundle;
+use crate::component::ComponentId;
+use crate::relationship::{RelatedSpawner, RelatedSpawnerCommands};
+use crate::system::EntityCommands;
+use crate::world::{DeferredWorld, EntityWorldMut};
+use crate::{
+    component::Component,
+    entity::{Entity, VisitEntities},
+    reflect::{
+        ReflectComponent, ReflectFromWorld, ReflectMapEntities, ReflectVisitEntities,
+        ReflectVisitEntitiesMut,
+    },
+    relationship::{Relationship, RelationshipSources},
+    world::{FromWorld, World},
+};
+use alloc::{format, string::String, vec::Vec};
+use bevy_ecs_macros::VisitEntitiesMut;
+use bevy_reflect::Reflect;
+use core::slice;
+use disqualified::ShortName;
+use log::warn;
+
+#[derive(Relationship, Clone, Reflect, VisitEntities, VisitEntitiesMut, PartialEq, Eq, Debug)]
+#[reflect(
+    Component,
+    MapEntities,
+    VisitEntities,
+    VisitEntitiesMut,
+    PartialEq,
+    Debug,
+    FromWorld
+)]
+#[relationship(relationship_sources = Children)]
+pub struct Parent(pub Entity);
+
+impl Parent {
+    pub fn get(&self) -> Entity {
+        self.0
+    }
+}
+
+// TODO: We need to impl either FromWorld or Default so Parent can be registered as Reflect.
```

Comment:
The pattern that users have come up with is to create a panicking constructor in the `Default` impl. I don't think that would work here though, since you need to actually create an instance.

We *could* store an `Option` inside, but that's a fair bit of UX pain.

Reply:
Note that this is carryover from the current hierarchy/scene code. I don't feel any particular inclination to sort this out here.